### PR TITLE
feat(sample): create example for using dynamodb data mapper annotations

### PR DESCRIPTION
**Pull request by CoreyCole**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?
There is no dynamoDB example, but there is one for mongoDB

## What is the new behavior?
This PR has a dynamoDB example. 

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
The `@aws/data-mapper-annotations` package is built on top of the AWS SDK. The introductory blog post with examples can be found [here](https://aws.amazon.com/blogs/developer/introducing-the-amazon-dynamodb-datamapper-for-javascript-developer-preview/) and the source code for the relevant packages can be found [here](https://github.com/awslabs/dynamodb-data-mapper-js/).

### CODE REVIEW

File: sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { Cat } from './schemas/cat.schema';
+
+@Controller('cats')
+export class CatsController {
+  constructor(private readonly catsService: CatsService) {}
+
+  @Post('createTable')
+  async createTable() {
+    await this.catsService.createCatsTable()
+      .catch(error => console.error(error));
```
Comment (LukvonStrom) [inline sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts]: I would suggest wrapping this in a try/catch block like this:
```
try{
await this.catsService.createCatsTable()
}catch(error){
 console.error(error);
}
```

File: sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    this.catsService.create(createCatDto)
+      .catch(error => console.error(error));
+  }
+
+  @Get()
+  async findAll(): Promise<Cat[]> {
+    return this.catsService.findAll()
+      .catch(error => {
+        console.error(error);
+        return [];
```
Comment (LukvonStrom) [inline sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts]: I would suggest the use of await here: 
```
try{
 return await this.catsService.findAll()
}catch(error){
 console.error(error);
 return [];
}
```

File: sample/20-dynamodb-data-mapper-annotations/src/cats/cats.service.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  async deleteCatsTable() {
+    await this.dynamoDB.mapper.deleteTable(Cat);
+  }
+
+  async create(createCatDto: CreateCatDto): Promise<Cat> {
+    const createdCat = new Cat();
+    createdCat.createdAt = new Date();
+    createdCat.name = createCatDto.name;
+    createdCat.age = createCatDto.age;
+    createdCat.breed = createCatDto.breed;
```
Comment (LukvonStrom) [inline sample/20-dynamodb-data-mapper-annotations/src/cats/cats.service.ts]:
```
const createdCat = new Cat();
createdCat.createdAt = new Date();
const {name, age, breed} = createCatDto;
Object.assign(createdCat, {name, age, breed});
```

File: sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts (hunk 2 above)
Comment (cojack) [reply sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts]: Even, better, point to the docs, where is explained, how to properly catch exception using exception interceptor.

Comment (LukvonStrom) [reply sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts]: Oh shoot, I totally forgot about that 🙈 that may be even better!

**Comment by CoreyCole** [comment]
@LukvonStrom any idea how to write a meaningful test for this?

**Comment by LukvonStrom** [comment]
@CoreyCole Not really at the Moment, sorry :/

File: sample/20-dynamodb-data-mapper-annotations/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+version: "3"
+
+services:
+  mongodb:
+    image: cnadiminti/dynamodb-local:latest
```
Comment (kamalgill) [inline sample/20-dynamodb-data-mapper-annotations/docker-compose.yml]: Why not reference the official AWS dynamodb-local image, per https://hub.docker.com/r/amazon/dynamodb-local/ ?

File: sample/20-dynamodb-data-mapper-annotations/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# NestJS with DynamoDB Data Matter Annotations
```
Comment (kamalgill) [inline sample/20-dynamodb-data-mapper-annotations/README.md]: Should be `# NestJS with DynamoDB Data Mapper Annotations` (note "Matter" typo)

File: sample/20-dynamodb-data-mapper-annotations/README.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+The `@aws/data-mapper-annotations` package is built on top of the AWS SDK. The introductory blog post with examples can be found [here](https://aws.amazon.com/blogs/developer/introducing-the-amazon-dynamodb-datamapper-for-javascript-developer-preview/) and the source code for the relevant packages can be found [here](https://github.com/awslabs/dynamodb-data-mapper-js/).
+
+#### Running the local example
+The local dynamoDB instance will be run using `docker-compose`. You'll need to install docker community edition. Then, you'll need to install node_modules (`npm i`).
```
Comment (kamalgill) [inline sample/20-dynamodb-data-mapper-annotations/README.md]: ...will be run using `docker-compose up`, ...

**Comment by kibertoad** [comment]
@CoreyCole This looks good! Any chance you could update to accomodate review comments so far?

**Comment by kamilmysliwiec** [comment]
Ping 🏓

**Comment by CoreyCole** [comment]
Thank you @kamalgill for the notes! I've updated my branch according to all of them.

Also, thank you for the ping- I missed this notification.

File: sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts (hunk 2 above)
Comment (CoreyCole) [reply sample/20-dynamodb-data-mapper-annotations/src/cats/cats.controller.ts]: added #a8ea4e9

**Comment by CoreyCole** [comment]
Travis is having a problem with the amazon docker image. I will revert that for now.

**Comment by kamilmysliwiec** [comment]
We don't plan to add more examples to this repo. Thank you for your contribution though

**Pull request closed without merging**